### PR TITLE
Add redshift dialect (WIP)

### PR DESCRIPTION
**Pull request by GeorgeSittas**
This is an attempt to add support for the redshift dialect. Feedback is welcome. Plan to add more features soon.

### CODE REVIEW

File: sqlglot/dialects/redshift.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            "TIMETZ": TokenType.TIMESTAMPTZ,
+            "VARBYTE": TokenType.VARBYTE,
+            "SIMILAR TO": TokenType.SIMILAR_TO,
+        }
+
+    class Generator(Postgres.Generator):
+        TYPE_MAPPING = {
+            **Postgres.Generator.TYPE_MAPPING,
+            exp.DataType.Type.BINARY: "VARBYTE",
+            exp.DataType.Type.GEOMETRY: "GEOMETRY",
```
Comment (tobymao) [inline sqlglot/dialects/redshift.py:54]: do geometry and hll sketch, or super needs these mappings? doesn't seem to do anything

File: sqlglot/dialects/redshift.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            "TIME": TokenType.TIMESTAMP,
+            "TIMETZ": TokenType.TIMESTAMPTZ,
+            "VARBYTE": TokenType.VARBYTE,
+            "SIMILAR TO": TokenType.SIMILAR_TO,
+        }
+
+    class Generator(Postgres.Generator):
+        TYPE_MAPPING = {
+            **Postgres.Generator.TYPE_MAPPING,
+            exp.DataType.Type.BINARY: "VARBYTE",
```
Comment (tobymao) [inline sqlglot/dialects/redshift.py:53]: why do you need a varbyte type? why not just in keywords varbyte -> Binary?

File: sqlglot/parser.py [hunk 3]
```diff
@@ -1547,7 +1554,10 @@ def _parse_types(self):
                 self.raise_error("Expecting >")
 
         if type_token in self.TIMESTAMPS:
-            tz = self._match(TokenType.WITH_TIME_ZONE)
+            tz = (
+                self._match(TokenType.WITH_TIME_ZONE)
+                or type_token == TokenType.TIMESTAMPTZ
```
Comment (tobymao) [inline sqlglot/parser.py:1559]: was this an old bug?

Comment (GeorgeSittas) [reply sqlglot/parser.py:1559]: I think so yeah, timestamptz would be parsed as timestamp.

File: sqlglot/dialects/redshift.py (hunk 2 above)
Comment (GeorgeSittas) [reply sqlglot/dialects/redshift.py:53]: Yeah I'll fix this. Redshift supports varbyte, varbinary and binary varying.

File: sqlglot/dialects/redshift.py (hunk 1 above)
Comment (GeorgeSittas) [reply sqlglot/dialects/redshift.py:54]: I think there are some use cases. For example:
- https://hevodata.com/learn/redshift-super-data-type/  (`CREATE TABLE region_nations_noshred (rdata SUPER);`)
- https://www.youtube.com/watch?v=f4EpjeuOchE (`CREATE TABLE ... (... active_users HLLSKETCH, ...)`
- https://docs.aws.amazon.com/redshift/latest/dg/GeometryType-function.html (not explicitly used).

But I also didn't find many other references, so I could leave them out for the time being.

**Review by tobymao — APPROVED** [review]

**Pull request merged**